app/models/planning_application.rb

```diff
@@ -746,16 +745,22 @@ def generate_document_tabs(tabs = Document::DEFAULT_TABS)
   end
 
   def environment_impact_assessment_status
-    case environment_impact_assessment
-    when TrueClass
-      :required
-    when FalseClass
-      :not_required
-    when NilClass
+    if environment_impact_assessment.present?
+      environment_impact_assessment.required ? :required : :not_required
+    else
       :not_started
+    end
+  end
+
+  def modify_expiry_date
+    if environment_impact_assessment.required?
+      self.expiry_date += 56.days
     else
-      raise ArgumentError, "Unexpected value for 'environment_impact_assessment': #{environment_impact_assessment.inspect}"
+      self.expiry_date = DAYS_TO_EXPIRE.days.after(validated_at || received_at)
+      self.target_date = 35.days.after(validated_at || received_at)
```

Comment:
Is this the same as `set_key_dates`. May want this to be consistent in case we change it in one place